source/adios2/toolkit/sst/cp/cp_writer.c

```diff
@@ -872,11 +872,19 @@ WS_ReaderInfo WriterParticipateInReaderOpen(SstStream Stream)
     CP_WSR_Stream->ReaderStatus = Opening;
     if (ReturnData->SpecPreload == SpecPreloadOn)
     {
-
-        CP_WSR_Stream->PreloadMode = SstPreloadSpeculative;
-        CP_WSR_Stream->PreloadModeActiveTimestep = 0;
-        CP_verbose(Stream, PerStepVerbose,
-                   "Setting SpeculativePreload ON for new reader\n");
+        if (strcmp(Stream->DP_Interface->DPName, "ucx") != 0)
```

Comment:
I think you can kill this change.  The RDMA dataplane also does not support speculative preload (only learned preload), but doesn't get handled specially.  This is reasonably in line with established ADIOS practice where unsupported engine parameters simply have no effect.  (This can be a pain, because it also means that misspelled engine parameters are silently ignored, but correcting that is another subject...)

Reply:
Thanks for your detailed explanation, dropped now.